haxepunk/graphics/BitmapText.hx

```diff
@@ -254,14 +227,9 @@ class BitmapText extends Graphic
 	function parseText():Void
 	{
 		// clear current opcode list
-		while (opCodes.length > 0)
-		{
-			opCodes.pop();
-		}
-		while (_scaleStack.length > 0)
-		{
-			_scaleStack.pop();
-		}
+		HXP.clear(opCodes);
```

Comment:
- The return type of splice is an array of the elements you removed (http://api.haxe.org/Array.html#splice). It has to allocate a new array for the return value.

  Not familiar enough with hxcpp source but I'm guessing it's here: https://github.com/HaxeFoundation/hxcpp/blob/master/src/Array.cpp#L928 (and on 924, even if the array was empty it still allocates a new empty array for the return value.)

- If you run with hxscout and just add things to an array and call splice repeatedly, you can verify that arrays are being allocated, and you'll see resulting GC spikes.